### PR TITLE
A11y: Added aria-label to ColorPicker component

### LINKED ISSUE #65851: ID 260407: [Programmatic Access - Simon - Thresholds]: Name is not defined properly for the "Undefined color" button under "Thresholds" controls.
### **User Experience:**  
Visually impaired and cognitive users who depend on screen reader will get misguided while navigating, if the proper name is not defined for 'Undefined color' button controls present on the page.
_Note: User credentials should NOT be included in the bug._

### **Repro Steps**: 

1. Open #Url: [play.grafana.org](https://play.grafana.org/) 
2. Sign in with valid credentials. "Home" page appears. 
3. Navigate the Feature Showcase pane heading and select it. 
4. Navigate the expanded Dropdown and select the "Edit" menu item. 
5. Navigate the page "Edit" page which appears. 
6. Navigate to the "All" buttons and select it. 
7. Navigate to "Thresholds" button and expand it. 
8. Navigate all the controls present under it. Verify the issue

**Actual Result:** 
Issue: `Name` is not defined properly for the "Undefined color" button under "Thresholds" controls. 

**Observation:** 
While navigating "Thresholds" screen reader focus moves at "Thresholds1" spin button announces as "Undefined color" button instead of "Red color".

**Expected Result:** 
`Name` should be properly defined as "Red color" button and screen reader should announce as " Red Color" button. Ex: Screen reader should announce as red color and green color.
_Note: Same issue exists with 'Thresholds2" base text box._

**### Severity:**
`SEV3`

**### Area:**
`ColorPicker` / `ColorSwatch Components` (Saga UI Design System)

### PR DESCRIPTION
**Pull request by RoxanaAnamariaTurc**
**What is this feature?**

Added `aria-label` to ColorPicker component.

**Why do we need this feature?**

To ensure visually impaired and cognitive users who depend on screen reader can navigate Grafana.

**Who is this feature for?**

Visually impaired and cognitive users who depend on screen reader

**Which issue(s) does this PR fix?**: 

Fixes #65851

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: packages/grafana-ui/src/components/ColorPicker/ColorPicker.tsx [hunk 1]
```diff
@@ -35,13 +35,12 @@ export const colorPickerFactory = <T extends ColorPickerProps>(
...
 
     render() {
-      const { theme, children, onChange } = this.props;
+      const { theme, children, onChange, color } = this.props;
       const styles = getStyles(theme);
       const popoverElement = React.createElement(popover, {
         ...{ ...this.props, children: null },
         onChange,
       });
-
```
Comment (JoaoSilvaGrafana) [inline packages/grafana-ui/src/components/ColorPicker/ColorPicker.tsx:44]: Can readd the space here

File: packages/grafana-ui/src/components/ColorPicker/ColorPicker.tsx [hunk 2]
```diff
@@ -35,13 +35,12 @@ export const colorPickerFactory = <T extends ColorPickerProps>(
     pickerTriggerRef = createRef<any>();
 
     render() {
-      const { theme, children, onChange } = this.props;
+      const { theme, children, onChange, color } = this.props;
```
Comment (tskarhed) [inline packages/grafana-ui/src/components/ColorPicker/ColorPicker.tsx:38]: ⭐ 👍

File: packages/grafana-ui/src/components/ColorPicker/ColorSwatch.tsx [hunk 3]
```diff
@@ -29,12 +29,17 @@ export const ColorSwatch = React.forwardRef<HTMLDivElement, Props>(
...
+    const colorLabel = ariaLabel || label;
 
     return (
       <div ref={ref} className={styles.wrapper} data-testid={selectors.components.ColorSwatch.name} {...otherProps}>
         {hasLabel && <span className={styles.label}>{label}</span>}
-        <button className={styles.swatch} {...focusProps} aria-label={colorLabel} type="button" />
+        <button
+          className={styles.swatch}
+          {...focusProps}
+          aria-label={colorLabel ? `${colorLabel} color` : 'Pick color'}
```
Comment (JoaoSilvaGrafana) [inline packages/grafana-ui/src/components/ColorPicker/ColorSwatch.tsx:40]:
```suggestion
          aria-label={colorLabel ? `${colorLabel} color` : 'Pick a color'}
```

**Review by JoaoSilvaGrafana — COMMENTED** [review]
looks good otherwise! 🎉

File: packages/grafana-ui/src/components/ColorPicker/ColorSwatch.tsx (hunk 3 above)
Comment (tskarhed) [inline packages/grafana-ui/src/components/ColorPicker/ColorSwatch.tsx:40]: Should the check here be `hasLabel`?

File: packages/grafana-ui/src/components/ColorPicker/ColorPicker.tsx [hunk 4]
```diff
@@ -72,7 +71,8 @@ export const colorPickerFactory = <T extends ColorPickerProps>(
                     ref={this.pickerTriggerRef}
                     onClick={showPopper}
                     onMouseLeave={hidePopper}
-                    color={theme.visualization.getColorByName(this.props.color || '#000000')}
+                    color={theme.visualization.getColorByName(color || '#000000')}
+                    aria-label={color}
```
Comment (tskarhed) [inline packages/grafana-ui/src/components/ColorPicker/ColorPicker.tsx:75]: Are there cases where these two should differ, or could you assign them to the same variable?

File: packages/grafana-ui/src/components/ColorPicker/ColorSwatch.tsx (hunk 3 above)
Comment (RoxanaAnamariaTurc) [reply packages/grafana-ui/src/components/ColorPicker/ColorSwatch.tsx:40]: Thanks, made the change

Comment (RoxanaAnamariaTurc) [reply packages/grafana-ui/src/components/ColorPicker/ColorSwatch.tsx:40]: Thanks, I updated the string

File: packages/grafana-ui/src/components/ColorPicker/ColorPicker.tsx (hunk 4 above)
Comment (RoxanaAnamariaTurc) [reply packages/grafana-ui/src/components/ColorPicker/ColorPicker.tsx:75]: The color that is passed to the ColorSwatch is actually a hex color and we want the aria-label to have the name of the color if there is one. That's why we have it like this.

File: packages/grafana-ui/src/components/ColorPicker/ColorPicker.tsx (hunk 1 above)
Comment (RoxanaAnamariaTurc) [reply packages/grafana-ui/src/components/ColorPicker/ColorPicker.tsx:44]: I checked and there isn't a white space in my code. I made the updates suggested earlier and will push this again now.

**Review by tskarhed — APPROVED** [review]

**Review by JoaoSilvaGrafana — APPROVED** [review]

**Pull request merged**